coalib/parsing/FilterHelper.py

```diff
@@ -75,8 +75,11 @@ def apply_filters(filters, bears=None, sections=None):
         items = sections
         applier = _apply_section_filter
 
-    for filter in filters:
-        filter_name, *filter_args = filter
+    for filter_name, *filter_args in filters.items():
+        for key in filter_args:
+            if isinstance(key, list):
+                filter_args = [
+                    item for sublist in filter_args for item in sublist]
```

Comment:
To convert a list of lists into a list. --> [['C', 'java']] ==> ['C', 'java']

Reply:
This would give the following result 
```
language
('c', 'java')
can_fix
('s', 'y', 'n', 't', 'a', 'x')
section_tags
('s', 'a', 'v', 'e')
```
instead of what I am aiming for 
```
language
['c', 'java']
can_fix
['syntax']
section_tags
['save']
```
I wrote this extra code to change the filter args from 
```
[['c', 'java']]
['syntax']
['save']
```
to 
```
['c', 'java']
['syntax']
['save']
```

Reply:
```py
>>> filters = [['langauge', 'c', 'java'], ['can_fix', 'syntax']]
>>> dict_filters = {}
>>> 
>>> for filter_name, *filter_args in filters:
...   filter_args = tuple(filter_args)
...   dict_filters[filter_name] = filter_args
... 
>>> 
>>> dict_filters
{'langauge': ('c', 'java'), 'can_fix': ('syntax',)}
>>> 
```

What am I doing wrong here?

Reply:
I changed filters from a list into a dict.
The new filters is 
```
{'language': ['c', 'java'], 'can_fix': 'syntax', 'section_tags': 'save'}
```

Reply:
Ok, so my understanding is you have already performed partial conversion to a dict based filter structure. A few remarks here:
- Use OrderedDicts, filtering order depends is important here.
- Use tuples for holding args, not lists.
- Do not partially convert the filters, you should rather have all the filter args as tuples. `{'language': ('c', 'java'), 'can_fix': ('syntax',), 'section_tags': ('save',)}` otherwise it will force all the other unrelated methods to handle this case.

Reply:
I would suggest doing this step wise to understand it better. Make everything work with tuple right now and then introduce dicts.